pkg/controller/agent/fleet.go

```diff
@@ -265,6 +285,30 @@ func maybeReconcileFleetEnrollment(params Params, result *reconciler.Results) En
 	return token
 }
 
+func getKibanaBasePath(ctx context.Context, client k8s.Client, kibanaNSN types.NamespacedName) (string, error) {
+	var kb v1.Kibana
+	err := client.Get(ctx, kibanaNSN, &kb)
```

Comment:
Does it mean that the Agent controller should watch Kibana resources? (I don't think it is the case, I might be wrong though)

Reply:
@barkbay this gets called inside the Fleet reconcile loop, and on every reconcile we will end up checking the basePath and updating if needed, in this case do we need a watch on Kibana resources ? would agent be using the API outside of the reconciliation loop, just trying to understand the workflow since I am not familiar with this code.

Reply:
🤦 , you're right, sorry for the noise.